test/jit/test_typing.py

```diff
@@ -73,11 +73,17 @@ def test_dict_tensor_key(a, t):
         self.checkScript(test_dict_tensor_key, (dict_a, inp1))
         self.checkScript(test_dict_tensor_key, (dict_a, inp2))
 
-    def test_dict_types(self):
-        with self.assertRaisesRegex(RuntimeError, "single type"):
-            @torch.jit.script
-            def foo():
-                new_item = {'score': [1.0], 'ys': [1, 2, 3]}
+    def test_dict_value_type_refinement_defaults_to_Any(self):
```

Comment:
Are other ways of constructing heterogenous-value dicts supported too? like from dict comprehension, from gradually inserting different types of values etc.

Reply:
I added some new logic and test cases. Please take a look at the test cases I commented out and let me know whether you think we should support that too

Reply:
cc @gmagogsfm Reminder to look at those commented-out test cases :)